massa-client/src/repl.rs

```diff
@@ -135,20 +136,50 @@ impl Completion for CommandCompletion {
     /// Simple completion implementation based on substring
     fn get(&self, input: &str) -> Option<String> {
         let input = input.to_string();
-        let suggestions: Vec<&str> = self
-            .options
-            .iter()
-            .filter(|s| s.len() >= input.len() && input == s[..input.len()])
-            .map(|s| &s[..])
-            .collect();
-        if !suggestions.is_empty() {
-            println!();
-            for suggestion in &suggestions {
-                println!("{}", style(suggestion).dim());
+        if input.contains(' ') {
+            let mut args: Vec<&str> = input.split(' ').collect();
+            let mut default_path = "./";
+            let path_to_complete = args.last_mut().unwrap_or(&mut default_path);
+            if let Ok(paths) = glob(&(path_to_complete.to_owned() + "*")) {
+                let suggestions: Vec<String> = paths
+                    .filter_map(|x| {
+                        if let Ok(path) = x {
+                            Some(path.display().to_string())
+                        } else {
+                            None
+                        }
+                    })
+                    .collect();
+                if !suggestions.is_empty() {
+                    println!();
+                    for path in &suggestions {
+                        println!("{}", style(path).dim())
+                    }
+                    *path_to_complete =
+                        longest_common_prefix(suggestions.iter().map(|s| &s[..]).collect());
+                }
+                Some(args.join(" "))
+            } else {
+                println!();
+                println!("Bad syntax for path completion.");
```

Comment:
I'm not sure how useful this line is for the end user

Reply:
It refer to the discussion above with damir but I don't know what to put as error message. Do you an idea ?

Reply:
I'd show nothing. Damir's concern was more about the client not crashing, not really having a nice error message. In a shell when path auto-completion fails, nothing happens ? I'd go for that. Maybe In a follow up, if you want to be extra fancy you could show the available file paths in the current directory even if they don't match after tab is pressed a second time.

Reply:
Ah ok so I remove it !